src/lib/annotations/doc/DocDrawingThread.js

```diff
@@ -1,78 +1,88 @@
     pageEl;
 
     //--------------------------------------------------------------------------
     // Public
     //--------------------------------------------------------------------------
     /**
+     * [constructor]
+     *
+     * @inheritdoc
+     * @param {AnnotationThreadData} data - Data for constructing thread
+     * @return {DocDrawingThread} Drawing annotation thread instance
+     */
+    constructor(data) {
+        super(data);
+
+        this.handleStart = this.handleStart.bind(this);
+        this.handleMove = this.handleMove.bind(this);
+        this.handleStop = this.handleStop.bind(this);
+        this.checkAndHandleScaleUpdate = this.checkAndHandleScaleUpdate.bind(this);
+        this.reconstructBrowserCoordFromLocation = this.reconstructBrowserCoordFromLocation.bind(this);
+    }
+    /**
      * Handle a pointer movement
      *
      * @param {Object} location - The location information of the pointer
      * @return {void}
      */
     handleMove(location) {
-        const pageChanged = this.lastPage && this.lastPage !== location.page;
-
-        this.lastPage = location.page;
-        if (!this.pageEl || pageChanged) {
-            this.pageEl = docAnnotatorUtil.getPageEl(this.annotatedElement, location.page);
-            if (pageChanged) {
-                this.handleStop(location);
-            }
+        const pageChanged = this.hasPageChanged(location);
+        if (this.drawingFlag !== STATES_DRAW.draw || pageChanged) {
             return;
         }
 
         const [x, y] = docAnnotatorUtil.getBrowserCoordinatesFromLocation(location, this.pageEl);
-        if (this.drawingFlag === STATES_DRAW.draw) {
-            this.pendingPath.addCoordinate(x, y);
-
-            // Cancel any pending animation to a new request.
-            if (this.lastAnimationRequestId) {
-                window.cancelAnimationFrame(this.lastAnimationRequestId);
-            }
-            // Keep animating while the drawing flag is down
-            this.lastAnimationRequestId = window.requestAnimationFrame(this.render);
-        }
+        const browserLocation = annotatorUtil.createLocation(x, y);
+        this.pendingPath.addCoordinate(location, browserLocation);
     }
 
     /**
      * Start a drawing stroke
      *
+     * @param {Object} location - The location information of the pointer
      * @return {void}
      */
-    handleStart() {
-        this.drawingFlag = STATES_DRAW.draw;
-        const scale = getScale(this.annotatedElement);
-        const context = docAnnotatorUtil.getContext(
-            this.pageEl,
-            CLASS_ANNOTATION_LAYER_DRAW,
-            PAGE_PADDING_TOP,
-            PAGE_PADDING_BOTTOM
-        );
+    handleStart(location) {
+        const pageChanged = this.hasPageChanged(location);
+        if (pageChanged) {
+            this.handleStop();
+            this.saveAnnotation();
+            this.emit('drawthreadcommited');
```

Comment:
For now, I will be using the model of soft commiting any drawings done on the page assigned to the drawing thread and creating a new thread assigned to the new page.